docs/ImageMirrorSetDesign.md

```diff
@@ -0,0 +1,167 @@
+# Summary
+
+Users need a way to set up mirror configuration. Users can set mirror configurations using ImageContentSourcePolicy currently via MCO currently, but if they want to use mirror when pulling image with tag instead of digest in the pull spec, they need to follow this [solution](https://access.redhat.com/solutions/4817401) to update the registries configuration via MCO. We can hopefully help customers by documenting CRDs with the `ImageDigestMirrorSet` and `ImageTagMirrorSet` tuneables and have the MCO use these tuneables when rendering the registries.conf files to Ignition/disk.
+
+# Motivation
+
+The MCO contains the logic to write the registries.conf configuration files to Ignition. When Ignition starts on a machine it writes this file to configure the mirrors.
```

Comment:
I think the summary captures the motivation pretty well already, so we don't have to explain the MCO's role here